src/com/google/javascript/jscomp/GlobalNamespace.java

```diff
@@ -1075,9 +1088,22 @@ boolean isSimpleStubDeclaration() {
     }
 
     boolean canCollapse() {
-      return !inExterns && !isGetOrSetDefinition() && (declaredType ||
+      boolean isAnnotatedNoCollapse = docInfo == null ?
+          false : docInfo.isNoCollapse();
+      if (!isAnnotatedNoCollapse) {
+        Ref ref = this.getDeclaration();
+        if (ref != null && ref.node != null) {
+          JSDocInfo info = NodeUtil.getBestJSDocInfo(ref.node);
+          if (info != null) {
+            isAnnotatedNoCollapse = info.isNoCollapse();
+          }
+        }
+      }
+
+      return !inExterns && !isGetOrSetDefinition() && !isAnnotatedNoCollapse &&
+          (declaredType ||
           (parent == null || parent.canCollapseUnannotatedChildNames()) &&
-          (globalSets > 0 || localSets > 0) &&
+          (globalSets > 0 || localSets > 0) && localSetsWithNoCollapse <= 0 &&
```

Comment:
It shouldn't ever be negative - that would be a bug. I'll change that.

localSetsWithNoCollapse was added for this test case: https://github.com/google/closure-compiler/pull/805/files#diff-f6ebc91c417a43a65475b9a83521bc54R772

The test case fails without that check.